Upmerge from 4.4-dev

laoneo opened the pull request:
Please hold on with the merge till we have merge all the pr's for 4.4 Alpha 2.

The last commit contains a dependency update, when needed I can revert it.

rajputanuj31 committed:
Added Cypress test for categories of banners , contacts and content (#40706)

* added test for categories of banners,contacts and content

* update describe

* Update Categories.cy.js

* Update Categories.cy.js

* Update Categories.cy.js

---------

Co-authored-by: Allon Moritz <allon.moritz@digital-peak.com>

laoneo committed:
Do deep search in system tests instead of hardcoded sorting (#40778)

* Do deep search in system tests instead of hardcoded sorting

* title

* api categories

Kaushik1216 committed:
Api test for tags (#40763)

* Api test for tags

* Api test for tags

* Create tag api test added

* CS corrected

* remove unnecessary lines

* cs corrected

* Update Tags.cy.js

* cs

---------

Co-authored-by: Allon Moritz <allon.moritz@digital-peak.com>

rajputanuj31 committed:
added api test for field-articles (#40782)

alikon committed:
[4][com_newsfeed] hits must be 0 on new item (#40758)

laoneo committed:
Merge branch '4.3-dev' into 4.4-dev

Kaushik1216 committed:
Added Test for Newsfeed in Backend (#40741)

* Added Test for Newsfeed in Backend

* correct Added Newsfeed pr for cs

* Added Test for Newsfeed in Backend

* correct Added Newsfeed pr for cs

Kaushik1216 committed:
Api Test For Contact Field (#40785)

* Api Test For Contact Field

* cs

brianteeman committed:
Update alternate.xml (#503) (#40772)

brianteeman committed:
[4.3] addfavicon (#40735)

brianteeman committed:
[4.3] Guided Tours search by note (#40607)

andyforrest committed:
Verify request links (#40934)

* Verify request links to user are correct

* removed console logs

* Fixed linting errors

* removed unnecessary line of code

laoneo committed:
Remove hardcoded catid's in tests (#40988)

* Remove hardcoded catid's in tests

* cs

* slash

Kaushik1216 committed:
Adding more test for categories (#40733)

* Adding more test for categories

* Update Categories.cy.js

* Update Categories.cy.js

* Create category to delete through the form

---------

Co-authored-by: Allon Moritz <allon.moritz@digital-peak.com>

laoneo committed:
Convert the page break plugin to service providers (#40743)

* Convert the page break plugin to service providers

* DEPLOY

* test

---------

laoneo committed:
Rename the smart search tests to finder (#41012)

laoneo committed:
Enable PHP 8.2 unit tests (#41013)

laoneo committed:
Fixes the unit tests (#41014)

* Fixes the tests

* more docs

* revert

* old name

* revert date tests

laoneo committed:
Catch null email addresses in PunycodeHelper (#41011)

brianteeman committed:
[4.4] Inclusive Naming abort (comments) (#41025)

We should be using inclusive naming throughout the cms not just in the language strings.

Changes "abort" to "stop" or "cancel" as appropriate,

### Rationale
https://inclusivenaming.org/word-lists/tier-1/abort/

### Test
Code review Only

### Documentation
New page will be added to manual.joomla.org shortly

### Note
This is part of a series of pull requests that are submitted separately to make reviewing easier

laoneo committed:
Update dependencies (#40783)

* Update dependencies

* update

* update npm

laoneo committed:
Merge branch '4.3-dev' into 4.4-dev

laoneo committed:
Merge branch '4.4-dev' into upmerge

laoneo committed:
Revert the dependencies

laoneo committed:
Update dependencies

laoneo committed:
Inject the database into the installer script when possible (#40724)

* Inject the database into the installer script when possible

* Proxy the database in the legacy installer class

laoneo committed:
Ensure a language is set in all views (#41021)

brianteeman committed:
[4.3]LogtypeField code comment (#41007)

brianteeman committed:
[4.3[] action logs module code comments (#41006)

richard67 commented:
This PR will also fix issue #40770 if the dependency update is not reverted.

brianteeman committed:
[4.3] ComponentDispatcher comments (#41005)

dgrammatiko committed:
[4.3] Better cache invalidation (#40893)

richard67 commented:
> Please hold on with the merge till we have merge all the pr's for 4.4 Alpha 2.
>
> The last commit contains a dependency update, when needed I can revert it.

@laoneo @HLeithner Please don't wait too long with making a decision about the dependency update and merging this PR. As long as there has not been made the upmerge, my tool for generating the lists of deleted files and folders will produce lots of false alarms which I have to sort out. Better merge this soon and later make again an upmerge when the few remaining PRs in the 4.4-dev branch have been merged because those few PR's have no impact on the deleted files and folders lists.

brianteeman committed:
[4.4] Inclusive Naming abort (language strings) (#41024)

We should be using inclusive naming throughout the cms not just in the language strings.

Changes "abort" to "error" or "cancel" as appropriate,

### Rationale
https://inclusivenaming.org/word-lists/tier-1/abort/

### Test
Code review Only

### Documentation
New page will be added to manual.joomla.org shortly

### Note
This is part of a series of pull requests that are submitted separately to make reviewing easier

laoneo committed:
Merge branch '4.3-dev' into 4.4-dev

laoneo committed:
Merge branch '4.4-dev' into upmerge

laoneo committed:
Merge branch '5.0-dev' into upmerge

richard67 committed:
Update to changes from PRs #40743 and #40783 (#41031)

mfleeson committed:
EOS - Plugin Disabling Quick Icon Display and adding correct text definitions (#40774)

* Disabling Quick Icon Display and adding correct text definitions and Joomla 5 links

Disabling Quick Icon Display and adding correct text definitions and Joomla 5 links

* Adding Special:MyLanguage to messageLinks

* Updating Joomla x.x to Joomla! x.x in language definitions and formatting

Changing language definitions to use Joomla!
Removing commented out code
Fixing indent.

* Removing ! from Language strings for Joomla!

Based on doc rules:
Rule: No Bang!
References to Joomla in language strings should not include the exclamation point unless it is at the end of a sentence that is intended to be exclamatory.

* Removing unnecessary field values, changing text back to correct non icon-new-tab

Not sure how but language definitions wound back. Here's the correct versions from my test server.
Also removed the extra fields that are now not needed as displaying a quickicon is no longer in the functionality requirements.

* Formatting fix

* Cleaning code

Removing unnecessary language definitions
Removing unneeded code

---------

Co-authored-by: Allon Moritz <allon.moritz@digital-peak.com>
Co-authored-by: Martin Carl Kopp <6154099+MacJoom@users.noreply.github.com>

richard67 committed:
Fix PHPCS error from PR #41031 (#41035)

laoneo committed:
Merge branch '4.4-dev' into upmerge

richard67 reviewed (approved):
Looks all right by review.

MacJoom committed:
Joomla! 4.4.0 Alpha 2

laoneo committed:
Merge commit '36e4e3080f5b4456273af08737b1782b8c24c3d4' into upmerge

richard67 commented on libraries/src/Version.php:
```diff
@@ -66,15 +66,15 @@ final class Version
      * @since  3.8.0
      */
-    public const EXTRA_VERSION = 'alpha2-dev';
+    public const EXTRA_VERSION = 'alpha2';
```
I think the changes in file "libraries/src/Version.php" should be reverted.

laoneo replied:
Good catch, that was the only file which had no conflict

laoneo committed:
Revert version.php
